### PR TITLE
fix: empty and restore over 1,000 items

### LINKED ISSUE #11057: Empty Recycle Bin strange behavior
### The bug

Trying to delete all the photo from the recycle bin but seems to be impossible.
I had deleted some elements that now are in the recycle bin and I have to press multiple times "Empty Recycle Bin" and refresh the page.
The photo comes from my mobile phone with auto backup on.
There is no file from an external library.

Press "Empty Recycle Bin" seems to delete all
image
image

Now refresh the page
image

Press again "Empty Recycle Bin"
image
image

Refresh again
image

Press again "Empty Recycle Bin"
image
image

Refresh again
image

Press again "Empty Recycle Bin"
image
image

Refresh again
image

### The OS that Immich Server is running on

Ubuntu 22.04.4 LTS

### Version of Immich Server

v1.108.0

### Version of Immich Mobile App

v1.108.0 build 148

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
#
# WARNING: Make sure to use the docker-compose.yml of the current release:
#
# https://github.com/immich-app/immich/releases/latest/download/docker-compose.yml
#
# The compose file on main may not be compatible with the latest release.
#

name: immich

services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    # extends:
    #   file: hwaccel.transcoding.yml
    #   service: cpu # set to one of [nvenc, quicksync, rkmpp, vaapi, vaapi-wsl] for accelerated transcoding
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    ports:
      - 2283:3001
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    # extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
    #   file: hwaccel.ml.yml
    #   service: cpu # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    volumes:
      - model-cache:/cache
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: docker.io/redis:6.2-alpine@sha256:328fe6a5822256d065debb36617a8169dbfbd77b797c525288e465f56c1d392b
    healthcheck:
      test: redis-cli ping || exit 1
    restart: always

  database:
    container_name: immich_postgres
    image: docker.io/tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
      POSTGRES_INITDB_ARGS: '--data-checksums'
    volumes:
      - ${DB_DATA_LOCATION}:/var/lib/postgresql/data
    healthcheck:
      test: pg_isready --dbname='${DB_DATABASE_NAME}' --username='${DB_USERNAME}' || exit 1; Chksum="$$(psql --dbname='${DB_DATABASE_NAME}' --username='${DB_USERNAME}' --tuples-only --no-align --command='SELECT COALESCE(SUM(checksum_failures), 0) FROM pg_stat_database')"; echo "checksum failure count is $$Chksum"; [ "$$Chksum" = '0' ] || exit 1
      interval: 5m
      start_interval: 30s
      start_period: 5m
    command: ["postgres", "-c" ,"shared_preload_libraries=vectors.so", "-c", 'search_path="$$user", public, vectors', "-c", "logging_collector=on", "-c", "max_wal_size=2GB", "-c", "shared_buffers=512MB", "-c", "wal_compression=on"]
    restart: always

volumes:
  model-cache:
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=./library
# The location where your database files are stored
DB_DATA_LOCATION=./postgres

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secret for postgres. You should change it to a random password
DB_PASSWORD=MY_RANDOM_PASSWORD_AS_SUGGESTED

# The values below this line do not need to be changed
###################################################################################
DB_USERNAME=postgres
DB_DATABASE_NAME=immich
```

### Reproduction steps

```bash
1. Go to the trash page
2. Press "Empty Recycle Bin"
3. Confirm
4. Refresh the page
```

### Relevant log output

```shell
Using docker logs immich_server there is no error.
```

Checking the chrome dev tools network tab I see the DELETE request:

Request URL: http://example.com:2283/api/assets
Request method: DELETE
Status Code: 204 No Content
Request Payload:
image

There is nothing in the response tab. Double click on the network line: {"message":"Cannot GET /api/assets","error":"Not Found","statusCode":404,"correlationId":"..."}

### Additional information

This problem seems to happend not only with the "Empty Recycle Bin" button.
Select one day from the trash page 
image

Press the "Delete Permanently" button
image
image

Refresh the page
image

and so on

### PR DESCRIPTION
**Pull request by jrasm91**
Fixes #11057

Also correctly reports the trash restore/empty count, which was previously only the number of assets loaded in the UI.

### CODE REVIEW

File: server/src/services/trash.service.ts [hunk 1]
```diff
@@ -16,53 +19,91 @@ export class TrashService {
...
-  async restore(auth: AuthDto): Promise<void> {
-    const assetPagination = usePagination(JOBS_ASSET_PAGINATION_SIZE, (pagination) =>
-      this.assetRepository.getByUserId(pagination, auth.user.id, {
-        trashedBefore: DateTime.now().toJSDate(),
-      }),
-    );
+  async restoreAll(auth: AuthDto): Promise<TrashResponseDto> {
+    let count = 0;
+
+    while (true) {
```
Comment (zackpollard) [inline server/src/services/trash.service.ts:38]: Can this be a do while loop instead?

**Review by danieldietzler — APPROVED** [review]
LGTM

**Review by zackpollard — APPROVED** [review]

**Comment by zackpollard** [comment]
LGTM

File: server/src/enum.ts [hunk 2]
```diff
@@ -182,6 +182,12 @@ export enum UserStatus {
   DELETED = 'deleted',
 }
 
+export enum AssetStatus {
+  ACTIVE = 'active',
+  TRASHED = 'trashed',
+  DELETED = 'deleted',
+}
+
```
Comment (mertalev) [inline server/src/enum.ts:190]: I'm confused. Why do we need this?

Comment (jrasm91) [reply server/src/enum.ts:190]: To immediately remove assets from the trash can while they are being deleted in the background.

Comment (mertalev) [reply server/src/enum.ts:190]: Meh, I wouldn't have added a status column just for that. You could have had a websocket event for "deletion in progress" with a spinner or similar.

**Pull request merged**